run/idp-sql/test/e2e_test_setup.yaml

```diff
@@ -0,0 +1,50 @@
+steps:
+
+- id: 'Add a Secret to Secret Manager'
+  name: 'gcr.io/cloud-builders/gcloud'
+  entrypoint: '/bin/bash'
+  args:
+    - '-c'
+    - |
+      sed -i "s/PROJECT_ID:REGION:INSTANCE/${_CLOUD_SQL_CONNECTION_NAME}/" postgres-secrets.json
+      sed -i "s/PASSWORD_SECRET/${_DB_PASSWORD}/" postgres-secrets.json
+      sed -i "s/\"DB_NAME\": \"postgres\"/\"DB_NAME\": \"${_DB_NAME}\"/" postgres-secrets.json
+      sed -i "s/\"DB_USER\": \"postgres\"/\"DB_USER\": \"${_DB_USER}\"/" postgres-secrets.json
+
+      gcloud secrets create ${_SERVICE}-secrets \
+        --replication-policy="automatic" \
+        --data-file=postgres-secrets.json || echo "Secret already exists."
+
+- id: 'Build Container Image'
+  name: 'gcr.io/cloud-builders/docker'
+  args: ['build', '-t', 'gcr.io/${PROJECT_ID}/${_SERVICE}:${_VERSION}', '.']
+
+- id: 'Push Container Image'
+  name: 'gcr.io/cloud-builders/docker'
+  args: ['push', 'gcr.io/${PROJECT_ID}/${_SERVICE}:${_VERSION}']
+
+- id: 'Deploy to Cloud Run'
+  name: 'gcr.io/cloud-builders/gcloud:latest'
+  entrypoint: /bin/bash
+  args:
+  - '-c'
+  - |
+    gcloud run deploy ${_SERVICE} \
```

Comment:
I don't think I've experience an error deploying unless it's my fault the service doesn't start. Since deployment failures seem rare, I would lean on not implementing retries.

Reply:
I don't know that I've seen a trend in which operations fail most on nightly tests, I've seen build, deploy, and delete all have issues. We can leave it at this for now, but I'm strongly recommending every API interaction have a retry.